.github/workflows/CI.yml

```
@@ -37,17 +37,20 @@ jobs:
           - 'nightly'
         group: [ 'short', 'long' ]
         os:
-          - ubuntu-latest
+          - [Linux, RPTU]
```

Can we do that (I am guessing not, but perhaps worth a try?)
```suggestion
          - [Linux, RPTU, ${{ matrix.group }}]
```

---

While this may not work, note that we use the `os` field for just one thing (I believe), namely in line 28: `runs-on: ${{ matrix.os }}`.

So in theory it should be possible to use a more complex expression there which takes both `matrix.os` and `matrix.group` into account. You may have to rethink what kind of string is put into `matrix.os`, though.

---

Perhaps change the `matrix.os` values to just plain `Linux` or `macOS` and then change `runs-on: ${{ matrix.os }}` to
```
runs-on: [ ${{ matrix.os }}, RPTU, ${{ matrix.group }}$ ]
```

---

What I really would want to do is:
- test if `matrix.os` is a string or a "list" (??)
- if it is a string, pass it on
- if it is a list, append "group" to the list and pass it on

However the GitHub documentation on [expression](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions) is... "sparse", to put it kindly. So I have no idea if any of these are even possible. They don't ever mention what happens if `os: [a, b]` is interpolated into an expression like `runs-on: ${{ matrix.os }}` -- is it treated as a string, or an object? If it is an object, you'd expect some functionality to interact with it, but if it's there, then it is either undocumented or I just can't find the documentation :-(

---

(OK, to be fair, the docs do suggest that it may be possible to write `${{ matrix.os[42] }}` to access the object at position 41 (0-indexed? 1-indexed?) of `matrix.os` if it is a list. But still no idea how to find out whether it is a list (which we could avoid if we just make sure it is always one) nor how to append to it.

---

in case of  `os: [one, dimensional, list]`, `matrix.os` is one of those items (`one`, or `dimensional`, or `list`) at any given time

in case of
```
os:
  - [first, part, of, 2d, list]
  - [second, part, of 2d, list]
  ...
```
`matrix.os` is an entire list item (`first, part, of, 2d, list`), as a yaml array.

---

OK, I  had to search for "array" and it does mention a bit more: we can also use `contains` to check if an array contains some element; and `join` to turn an array into a list. Then there is also `fromJSON` and `format`.... So one could do this evil thing: take the list, turn it into a string via `join` using commas as seperators; use format to "append" to this flattened list, and put `[` and `]` around it. This now is valid JSON and we can use `fromJSON` to turn it back into the desired list with one element added.

Of course we will also be top on the list of people to be exterminated when the machines rise.

---

```
runs-on: [ ${{ matrix.os }}, RPTU, ${{ matrix.group }}$ ]
```
This idea doesn't work, for reasons I don't understand.

Please look at the CI run for fd433b64cd : `runs-on: [ '${{ matrix.os }}', 'high-memory']` makes it just ignore the config and run on `Linux, RPTU` anyway? Here, `matrix.os` should just be `Linux`, and `matrix.hosted` is `RPTU`, but its just defined, and not used anywhere....

https://github.com/oscar-system/Oscar.jl/actions/runs/11705888151